src/QsCompiler/Core/Transformations/TypeTransformation.fs

```diff
@@ -144,5 +144,5 @@ type TypeTransformationBase(options: TransformationOptions) =
                 | ExpressionType.Pauli -> this.OnPauli()
                 | ExpressionType.Range -> this.OnRange()
 
-            let ResolvedType t = ResolvedType.New(true, t)
-            ResolvedType |> Node.BuildOr t transformed
+            let range = this.OnRangeInformation t.Range
```

Comment:
Please make sure the order in which individual items in a node are processed matches other transformations; For the other nodes, ranges are processed before the kinds and the types.